tsdb/head_wal.go

```diff
@@ -236,6 +236,7 @@ Outer:
 			seriesPool.Put(v)
 		case []record.RefSample:
 			samples := v
+			minValidTime := h.minValidTime.Load()
 			// We split up the samples into chunks of 5000 samples or less.
 			// With O(300 * #cores) in-flight sample batches, large scrapes could otherwise
 			// cause thousands of very large in flight buffers occupying large amounts
```

Comment:
since we remove the comment about dropping the samples from the later block, should we move it here instead?

Reply:
It’s on line 254.